runtime/frequency/src/lib.rs

```diff
@@ -159,9 +159,27 @@ impl_opaque_keys! {
 	}
 }
 
+// The duplicate macros are annoying, but #[sp_version::runtime_version]
+// has fairly string limits on what can go in there.
+
+// Override the spec name when not mainnet to be frequency-rococo
+#[cfg(not(feature = "frequency"))]
+macro_rules! spec_name {
+	( $y:expr ) => {{
+		create_runtime_str!("frequency-rococo")
+	}};
+}
+
+#[cfg(feature = "frequency")]
+macro_rules! spec_name {
+	( $y:expr ) => {{
+		create_runtime_str!($y)
+	}};
+}
+
 #[sp_version::runtime_version]
 pub const VERSION: RuntimeVersion = RuntimeVersion {
-	spec_name: create_runtime_str!("frequency"),
+	spec_name: spec_name!("frequency"),
```

Comment:
Has to be a macro and has to have one param based on `#[sp_version::runtime_version]`